### ISSUE TITLE
How can I remove a failed battery-powered node that is marked as asleep?

### ISSUE THREAD

**Issue by danielduerr** [AlCalzone/node-zwave-js#6062]
I have a ZSE42 that is truly dead -- it fried or something.  It shows as "asleep", is not marked as failed, and so "Removed Failed Node" does not work because it doesn't know it is dead yet, and insists on pinging it first.

**Comment by kpine** [comment]
> "Removed Failed Node" does not work because it doesn't know it is dead yet, and insists on pinging it first.

Did you actually observe this behavior, or just assuming? You can remove any battery device with "Remove Failed Node", even those that are still part of the network.

**Comment by danielduerr** [comment]
Yes, I have observed this behavior. When I try to Remove Failed Node, it
never does anything. I’ve use the same function in other cases with powered
devices and it always works as expected. But with this battery powered
device, it never proceeds to remove it. And in the log I see a message that
it tried pinging it and it just does nothing after that.

**Comment by kpine** [comment]
Then I'd suggest providing debug logs of the removal attempt and the node debug info.

**Comment by danielduerr** [comment]
Ok will do.

Is there some amount of time that just needs to pass for the device to be
considered dead vs asleep? It seems like it’d remove as expected if it was
considered dead.

**Comment by kpine** [comment]
Battery devices are never considered dead. Only awake or asleep.

https://zwave-js.github.io/node-zwave-js/#/api/node?id=status

**Comment by danielduerr** [comment]
Got it. That is what I thought. And given this has always worked for me in
the past, and assumed it was behaving different due to being the first
battery device I’ve had to do this on.

Will you want me to enable any special debug logging when I send you the
logs?

**Comment by kpine** [comment]
Just follow the instructions here, using Debug level. https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=generating-log-files

Attach the log files to the issue.

**Comment by danielduerr** [comment]
Here's the node debug info:
```
{
  "id": 44,
  "name": "",
  "loc": "",
  "values": [],
  "groups": [],
  "neighbors": [],
  "ready": false,
  "available": true,
  "hassDevices": {},
  "failed": false,
  "inited": false,
  "eventsQueue": [
    {
      "time": "2023-07-21T22:36:36.574Z",
      "event": "sleep",
      "args": [
        0
      ]
    },
    {
      "time": "2023-07-21T22:36:36.574Z",
      "event": "sleep",
      "args": [
        0
      ]
    }
  ],
  "status": "Asleep",
  "interviewStage": "ProtocolInfo",
  "priorityReturnRoute": {},
  "customReturnRoute": {},
  "customSUCReturnRoutes": [],
  "applicationRoute": false,
  "hexId": "0xXXXX-0xXXXX-0xXXXX",
  "dbLink": "https://devices.zwave-js.io/?jumpTo=0xXXXX:0xXXXX:0xXXXX:0.0",
  "productLabel": "Unknown product 0xXXXX",
  "productDescription": "0xXXXX",
  "manufacturer": "Unknown manufacturer 0xXXXX",
  "protocolVersion": 3,
  "nodeType": 1,
  "endpointsCount": 0,
  "endpoints": [
    {
      "index": 0,
      "label": "Root Endpoint"
    }
  ],
  "isSecure": false,
  "security": "None",
  "supportsSecurity": false,
  "supportsBeaming": true,
  "isControllerNode": false,
  "isListening": false,
  "isFrequentListening": false,
  "isRouting": true,
  "keepAwake": false,
  "maxDataRate": 100000,
  "deviceClass": {
    "basic": 4,
    "generic": 7,
    "specific": 1
  },
  "lastActive": 1689957298906,
  "firmwareCapabilities": {
    "firmwareUpgradable": false
  },
  "deviceId": "undefined-undefined-undefined",
  "_name": "NodeID_44",
  "statistics": {
    "commandsTX": 0,
    "commandsRX": 0,
    "commandsDroppedRX": 0,
    "commandsDroppedTX": 0,
    "timeoutResponse": 0
  },
  "prioritySUCReturnRoute": false,
  "errorReceive": false,
  "errorTransmit": false
}
```

**Comment by danielduerr** [comment]
And here are the Silly logs which capture 2 attempts to remove the failed node...
[Logs.zip](https://github.com/zwave-js/zwave-js-ui/files/12135165/Logs.zip)

**Comment by kpine** [comment]
Strange, it does look like it does a ping and never recognizes that it fails (NoAck was received). You'll have to wait for someone (Al) with more knowledge to take a look.

Do you see any errors printed in the console (i.e. Docker logs)? Sometimes exceptions don't go in the file logs.

**Comment by danielduerr** [comment]
I agree it is weird.  FWIW I've done full power off/power ons including removing the USB stick and putting it back in after 30 seconds.  I feel like something odd is going on with my Zooz ZST10 stick, but I can't put my finger on it.

**Comment by kpine** [comment]
I don't think it has anything to do with the controller or other hardware. The driver software is not attempting to remove the node for whatever reason, TBD.

If you are desperate (e.g. it's causing issues with your network), you can probably remove it with the Windows PC Controller software.

**Comment by danielduerr** [comment]
Got it, thanks.  I can live with the unknown device for the time being.  But what is concerning me now is more around what seems like inability to properly exclude/include other devices.  For example, I tried excluding and then including a receptacle that had previously been working.  And either through smart start, or classic inclusion, it doesn't seem to want to come back in.  If my sense is correct here, then the inability to remove this failed device could be related.

**Comment by dfiore1230** [comment]
1 i am seeing the same issue with a BRK Brands ZCOMBO smoke detector (v1 not listed in the db)

```
{
  "id": 95,
  "name": "Guest Bedroom ZCOMBO",
  "loc": "",
  "values": [],
  "groups": [],
  "neighbors": [
    5,
    7,
    8,
    9,
    12,
    13,
    14,
    15,
    16,
    22,
    29,
    32,
    38,
    42,
    46,
    47,
    48,
    51,
    66,
    67
  ],
  "ready": false,
  "available": true,
  "hassDevices": {},
  "failed": false,
  "inited": false,
  "eventsQueue": [
    {
      "time": "2023-07-21T16:21:08.006Z",
      "event": "sleep",
      "args": [
        0
      ]
    }
  ],
  "status": "Asleep",
  "interviewStage": "ProtocolInfo",
  "priorityReturnRoute": {},
  "customReturnRoute": {},
  "customSUCReturnRoutes": [],
  "applicationRoute": false,
  "hexId": "0xXXXX-0xXXXX-0xXXXX",
  "dbLink": "https://devices.zwave-js.io/?jumpTo=0xXXXX:0xXXXX:0xXXXX:0.0",
  "productLabel": "Unknown product 0xXXXX",
  "productDescription": "0xXXXX",
  "manufacturer": "Unknown manufacturer 0xXXXX",
  "protocolVersion": 3,
  "nodeType": 1,
  "endpointsCount": 0,
  "endpoints": [
    {
      "index": 0,
      "label": "Root Endpoint"
    }
  ],
  "isSecure": false,
  "security": "None",
  "supportsSecurity": false,
  "supportsBeaming": true,
  "isControllerNode": false,
  "isListening": false,
  "isFrequentListening": false,
  "isRouting": true,
  "keepAwake": false,
  "maxDataRate": 40000,
  "deviceClass": {
    "basic": 4,
    "generic": 161,
    "specific": 0
  },
  "lastActive": null,
  "firmwareCapabilities": {
    "firmwareUpgradable": false
  },
  "deviceId": "undefined-undefined-undefined",
  "statistics": {
    "commandsTX": 0,
    "commandsRX": 0,
    "commandsDroppedRX": 0,
    "commandsDroppedTX": 0,
    "timeoutResponse": 0
  },
  "_name": "Guest Bedroom ZCOMBO",
  "prioritySUCReturnRoute": false
}
```

```
2023-07-24T14:08:11.877Z CNTRLR » [Node 095] pinging the node...
2023-07-24T14:08:11.883Z DRIVER   all queues busy
2023-07-24T14:08:11.887Z SERIAL » 0x010d00a9015f01002500000000f0d1                                    (15 bytes)
2023-07-24T14:08:11.888Z DRIVER » [Node 095] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      240
                                  └─[NoOperationCC]
2023-07-24T14:08:11.892Z SERIAL « [ACK]                                                                   (0x06)
2023-07-24T14:08:11.895Z SERIAL « 0x010401a90152                                                       (6 bytes)
2023-07-24T14:08:11.896Z SERIAL » [ACK]                                                                   (0x06)
2023-07-24T14:08:11.898Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2023-07-24T14:08:16.035Z SERIAL « 0x011d00a9f0010197007f7f7f7f7f0101070000000002040000000000000052    (31 bytes)
2023-07-24T14:08:16.037Z SERIAL » [ACK]                                                                   (0x06)
2023-07-24T14:08:16.039Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            240
                                    transmit status:        NoAck, took 4070 ms
                                    routing attempts:       4
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    TX channel no.:         1
                                    TX power:               0 dBm
                                    measured noise floor:   0 dBm
2023-07-24T14:08:16.043Z DRIVER   all queues idle
2023-07-24T14:08:21.045Z DRIVER   all queues busy
2023-07-24T14:08:21.048Z SERIAL » 0x0103003bc7                                                         (5 bytes)
2023-07-24T14:08:21.049Z DRIVER » [REQ] [GetBackgroundRSSI]
2023-07-24T14:08:21.053Z SERIAL « [ACK]                                                                   (0x06)
2023-07-24T14:08:21.055Z SERIAL « 0x0107013ba19b9b0162                                                 (9 bytes)
2023-07-24T14:08:21.056Z SERIAL » [ACK]                                                                   (0x06)
2023-07-24T14:08:21.058Z DRIVER « [RES] [GetBackgroundRSSI]
                                    channel 0: -95 dBm
                                    channel 1: -101 dBm
                                    channel 2: -101 dBm
2023-07-24T14:08:21.062Z DRIVER   all queues idle
```

**Comment by AlCalzone** [comment]
Seems to be limited to sleeping devices. I just tried with a mains-powered one and there it works.

**Comment by AlCalzone** [comment]
Can confirm. The `ping()` call never resolves for sleeping nodes, so this doesn't ever continue to removing.

**Comment by danielduerr** [comment]
Thank you @AlCalzone !

**Comment by AlCalzone** [comment]
released in 11.6.0 - now it's just a matter of time that applications pick it up.
